refactor(vue): watch for SingleList.jsx

ShahAnuj2610 opened the pull request:
## What this PR does?
Fix props getting updated in Redux store

## Test
https://www.loom.com/share/20eabb8891fa4abf8543494d43e650ac

bietkul commented on packages/vue/src/components/list/SingleList.jsx:
```diff
@@ -132,6 +121,17 @@ const SingleList = {
+		$props: {
+			deep: true,
+			handler(newVal) {
+				this.updateComponentProps(this.componentId, newVal, componentTypes.singleList);
```
@ShahAnuj2610 What is `newVal` here?
Can you log it and share a video?
Also for the performance reasons only update the valid props.

ShahAnuj2610 replied:
@bietkul 
> What is newVal here?

it's the new value for props. Will share the video.
> Also for the performance reasons only update the valid props

it's already handled in `updateComponentProps`
https://github.com/appbaseio/reactivecore/blob/c6deb6e40da8da8eb2ab7b1397ce838098c61e58/src/actions/props.js#L35
https://github.com/appbaseio/reactivecore/blob/c6deb6e40da8da8eb2ab7b1397ce838098c61e58/src/actions/props.js#L4

ShahAnuj2610 committed:
refactor(vue): watch for SingleList.jsx

## What this PR does?
fixes props getting updated in redux store

## Test
https://www.loom.com/share/20eabb8891fa4abf8543494d43e650ac
